### PR TITLE
CXP-2349: Code health updates for kafka-connect-s3

### PR DESCRIPTION
**Pull request by sugarcrm-jgminder**
- [x] Configure automatic code formatting via the spotless plugin
- [x] Perform the initial formatting of code (Google style)
- [x] Deprecate the `metrics/datadog` folder
- [x] Configure the `maven-publish` plugin to publish to CXP Nexus (i.e. via `gradle publish`)
- [ ] Add Dependabot support (requires maintainer privileges)
- [x] Use version constraints to manage dependencies

Note: Publishing to Nexus is still manual due to Nexus existing behind the company VPN. Users can set the following environment variables to publish locally:

- `MAVEN_USERNAME`
- `MAVEN_PASSWORD`

### CODE REVIEW

File: .github/workflows/gradle-build.yml [hunk 1]
```diff
@@ -22,4 +22,7 @@ jobs:
         uses: gradle/gradle-build-action@v2
 
       - name: Run Tests
+        env:
+          MAVEN_USERNAME: user
+          MAVEN_PASSWORD: password
         run: ./gradlew test
```
Comment (sugarcrm-jgminder) [inline .github/workflows/gradle-build.yml:28]: The `MAVEN_USER` and `MAVEN_PASSWORD` are set to dummy values so gradle can validate the values successfully without error.

**Review by ramanenka — APPROVED** [review]

File: settings.gradle [hunk 2]
```diff
@@ -5,7 +5,3 @@ include 'common'
 include 'sink'
 include 'source'
 include 'system_test'
-
-include ':metrics-datadog'
-project(':metrics-datadog').projectDir = new File('./metrics/datadog')
```
Comment (morozov) [inline settings.gradle:10]: > Deprecate the `metrics/datadog` folder

Should this code be deleted entirely? If it's not tested and is unused, there's no point in keeping it.

Comment (sugarcrm-jgminder) [reply settings.gradle:10]: It was deleted as a part of the same commit. We don't utilize DataDog so there's no point in maintaining this code.

Comment (morozov) [reply settings.gradle:10]: Oh, I didn't realize that those were just the two deleted files. Thanks.

I got misled by the "deprecate" wording. Usually, it means that the functionality is still supported but its usage is discouraged.

**Pull request merged**